backend/de.metas.business/src/main/java/de/metas/invoice/service/impl/AbstractInvoiceDAO.java

```diff
@@ -503,6 +503,32 @@ public <T extends org.compiere.model.I_C_Invoice> List<T> getByDocumentNo(final
 				.list(modelClass);
 	}
 
+	@Override
+	public ImmutableSet<I_C_Invoice> retrieveUnpaid(
```

Comment:
* pls rename param `docNos` to `onlyDocumentNos`
* pls rename param `docStatuses` to `onlyDocStatuses`
* pls introduce and use param `QueryLimit limit` to make sure we are not retrieving the whole database of unpaid invoices.